### ISSUE TITLE
Can't delete multiple Snapcast entries

### ISSUE THREAD

**Issue by duhow** [music-assistant/support#3345]
### What version of Music Assistant has the issue?

2.4.0b7

### What version of the Home Assistant Integration have you got installed?

2024.12.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Multiple Snapcast clients with same hostname, different MAC address.
Attempting to cleanup those manually after they timeout (disabled), but they reappear again, looks like delete is not working.

### How to reproduce

- Create multiple Snapcast clients with autodiscovery, same name, different MAC addr.
- This creates multiple entries in **MA Players**, each unique with its MAC .
- Wait several hours for timeout
- Entries are disabled
- Attempt to delete the Snapcast clients from **MA Players**, they disappear from the UI, but after reloading website they are still present. Also present in `server.json` configuration file of Snapserver.

### Music Providers

Not related.

### Player Providers

Snapcast

### Additional information

**Workaround:** Get to the container, delete `server.json` file, and restart Music Assistant server.

### What version of Home Assistant Core are your running

2024.12.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
You don’t mention where you are trying to delete them from.

**Comment by duhow** [comment]
Everything described is in MA .

**Comment by OzGav** [comment]
MA just shows the players supplied by the snapserver. If they are still present in the Snapweb/Snapclient UI then what you see is to be expected. 

Are you deleting them using this menu item in the MA UI? 
![image](https://github.com/user-attachments/assets/37061acd-66fc-48f0-ba46-dcfd19fc6c67)

**Comment by duhow** [comment]
Yes, I was attempting to delete like that. And it did not work. The file `server.json` (provided by Snapserver) still contained all the entries.

**Comment by OzGav** [comment]
Works fine for me. Are you using the MA snapserver? If not try with that.

**Comment by duhow** [comment]
I am using MA snapserver and had this issue.

**Comment by OzGav** [comment]
@SantiagoSotoC

**Comment by SantiagoSotoC** [comment]
You have to delete it from snapcast. snapweb should let you do it, then you can delete it from MA

**Comment by duhow** [comment]
Being unable to do it via MA sounds like a bug to me.

**Comment by SantiagoSotoC** [comment]
I thought you were using an external server. 
Yes, it seems to be a bug, most probably in the library we use, I would like more time to rewrite all that library, for now it has to wait a little bit.

**Comment by SantiagoSotoC** [comment]
@duhow Can you confirm if it will be fixed in the next update ?

**Comment by OzGav** [comment]
@duhow 2.5.0 has been released. This will be closed soon assuming fixed

**Comment by OzGav** [comment]
I confirm this has been fixed in 2.5.0
